compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIELowerToAIE.cpp

```diff
@@ -913,6 +816,7 @@ LogicalResult lowerToAIE(ModuleOp moduleOp) {
       if (isa<func::FuncOp, func::ReturnOp>(op)) {
         return WalkResult::advance();
       } else if (auto workgroupOp = dyn_cast<AMDAIE::WorkgroupOp>(op)) {
+        // TODO(newling) should be 1 device op per workgroup op, surely?
```

Comment:
Ideally yes, but not sure whether the lowering works with multiple device ops though and I think it doesn't.

Reply:
Ok, future work :)